rmf_site_editor/src/site/deletion.rs

```diff
@@ -114,14 +118,69 @@ impl Plugin for DeletionPlugin {
     }
 }
 
-fn handle_deletion_requests(mut deletions: EventReader<Delete>, mut params: DeletionParams) {
+#[derive(Deref, DerefMut)]
+pub struct DeletionBox(pub BoxedSystem<HashSet<Delete>, HashSet<Delete>>);
+
+#[derive(Default, Resource)]
+pub struct DeletionFilters {
+    boxed_systems: Vec<DeletionBox>,
+    pending_insertion: Vec<DeletionBox>,
+}
+
+impl DeletionFilters {
+    pub fn insert(&mut self, filter: DeletionBox) {
+        self.pending_insertion.push(filter);
+    }
+
+    fn insert_boxes(&mut self, world: &mut World) {
+        while !self.pending_insertion.is_empty() {
```

Comment:
This is a bit odd for what is essentially a for loop, you can refactor it with `drain` that will consume the elements of the vector instead of having to do a while / pop + unwrap as such:

```
for mut inserted in self.pending_insertion.drain(..) {
    [...]
}

``

Reply:
[10bca4e](https://github.com/open-rmf/rmf_site/pull/243/commits/10bca4e555a14f28a8fd7b455ccaeb9302ef6e93), along with Grey's suggestion for removing `clone()`